src/BlazorWasmSdk/Tasks/ComputeBlazorPublishAssets.cs

```diff
@@ -193,6 +194,22 @@ public override bool Execute()
                     {
                         newDotNetJs = new TaskItem(Path.GetFullPath(aotDotNetJs.ItemSpec), asset.CloneCustomMetadata());
                         newDotNetJs.SetMetadata("OriginalItemSpec", aotDotNetJs.ItemSpec);
+
+                        var existingRelativePath = newDotNetJs.GetMetadata("RelativePath");
+                        var existingRelativePathWithoutExtension = Path.Combine(Path.GetDirectoryName(existingRelativePath), Path.GetFileNameWithoutExtension(existingRelativePath));
+
+                        // A Base64 encoded hash can contain `/`, `=` and `+` which may be problematic for URLs/file names.
+                        // As we're only using the hash for the purposes of cache busting, we can opt to simply remove those chars 
+                        // from the hash (at the cost of a tiny bit of entropy).
+                        //
+                        // URL encoding is not an option as encoded file names are decoded at another layer and not
```

Comment:
Out of curiosity, what's url decoding these names?

Reply:
Not entirely sure, didn't want to spend too much time tracking that down given it's not really necessary (especially given we're willing to take a substring anyways).

Reply:
The final asset would not contain the symbols.

Assets are always copied to `Concat(BasePath, RelativePath)`. Otherwise is a bug

Reply:
> final asset would not contain the symbols

Yep, I'm trimming out the symbols below.

Reply:
> The final asset would not contain the symbols.

Yep, I'm trimming out the symbols below.

Reply:
Actually, use this https://github.com/dotnet/sdk/blob/main/src/RazorSdk/Tasks/ComputeCssScope.cs#L43-L71 (an adapted version)

Reply:
We did all the calculations at the time for the entropy and the risk of collisions and so on, so its better

Reply:
Updated.